.github/workflows/build.yml

```diff
@@ -17,17 +20,39 @@ on:
       - "**.cabal"
       - "**.yaml"
       - "**.yml"
+      - "**.toml"
+      - "**.rs"
+      - "**.wit"
 
 jobs:
-  build:
+  stack_build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: haskell/actions/setup@v2
       - run: stack build
-  test:
+  stack_test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: haskell/actions/setup@v2
       - run: stack test
+  cargo_run:
```

Comment:
It should be in an entirely different file, than they can have different file trigger pattern

Reply:
done